src/react/ISV/hooks/useIAMUser.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation, useQuery } from 'react-query';
+import { useMemo, useState } from 'react';
+import { useIAMClient } from '../../IAMProvider';
+import { useAssumeRoleQuery } from '../../next-architecture/ui/S3ClientProvider';
+
+interface IAMUser {
+  id: string;
+  name: string;
+}
+
+interface UseIAMUserProps {
+  IAMUserName: string;
+  IAMUserNameType: 'create' | 'existing';
+  onIAMUsersLoaded?: (users: IAMUser[]) => void;
+  onShouldGenerateKey?: (shouldGenerate: boolean) => void;
+}
+
+export const useIAMUser = ({
+  IAMUserName,
+  IAMUserNameType,
+  onIAMUsersLoaded,
+  onShouldGenerateKey,
+}: UseIAMUserProps) => {
+  const IAMClient = useIAMClient();
+  const { getQuery } = useAssumeRoleQuery();
+  const [users, setUsers] = useState<IAMUser[]>([]);
+  const [accessKeys, setAccessKeys] = useState<string[] | null>(null);
+
+  useQuery(
```

Comment:
Better use the full object approach with queryKey and queryFn because react-query 5+ deprecated this way of definining a query I think